src/app.js

```diff
@@ -1,48 +1,47 @@
-export default function (options) {
-	var model = options.model
-	var view = options.view
+export default function (app) {
+	var view = app.view || function () {
+		return ""
+	}
 
+	var model
 	var actions = {}
-	var effects = options.effects || {}
-	var reducers = options.reducers || {}
-	var subs = options.subscriptions
-
-	var router = options.router
-
+	var reducers
+	var effects
+	var subscriptions = []
+	var hooks = []
+	var plugins = app.plugins
 	var node
 	var root
 	var batch = []
 
-	var hooks = merge({
-		onAction: Function.prototype,
-		onUpdate: Function.prototype,
-		onError: function (err) {
+	var onError = function (err) {
+		if (!hook("onError", function (fn) {
+			fn(err)
+		})) {
 			throw err
 		}
-	}, options.hooks)
+	}
 
-	initializeActions(actions, effects, false)
-	initializeActions(actions, reducers, true)
+	use(app)
 
-	domContentLoaded(function () {
-		root = options.root || document.body.appendChild(document.createElement("div"))
+	for (var key in plugins) {
```

Comment:
@lukejacksonn for in is used to loop through props of an object. This can be used to traverse an array too, but it's a "hack". The best practice to traverse arrays is using a regular for loop or forEach.

I've changed them all to use a regular for loop, so say bye to for in! :)